src/daemon/normalize.js

```diff
@@ -0,0 +1,12 @@
+const punycode = require('punycode') // eslint-disable-line node/no-deprecated-api
+
+module.exports = function normalize(name) {
+  return name
+    ? punycode
+        .encode(name)
+        .toLowerCase()
+        .replace(/[^a-z0-9-.]/g, '-')
+        .replace(/-+/g, '-')
+        .replace(/-$/, '')
```

Comment:
Not sure to understand what they're for?

```js
.replace(/-+/g, '-')
.replace(/-$/, '')
```

Reply:
It’s to ensure that multiple invalid characters only get replaced by one `-`, and to remove trailing `-`s.

Reply:
Yes, this is just for tidiness sake. In the examples I posted server names include whitespace followed by non-alpha-numeric characters resulting in double hyphens.

It looks a little cleaner to me to tidy them, but the code I wrote mostly applies to my own naming scheme. For example, it doesn't remove leading hyphens, but that's only because I'd never named a server something that would result in a leading hyphen.

I'm happy to simplify this entire section and simply use the pure punycode replacements. That seems less likely to result in id collisions where multiple similarly-named servers normalize to the same server id.